### PR TITLE
Review 2024.3 documentation changes

### PR DESCRIPTION
**Pull request by seanbudd**
**Must be a squash merge**

Changes from this PR to the previous release:
- Files to check: `user_docs/en/userGuide.md`, `user_docs/en/changes.md` `developerGuide.md`
- [Compare this branch to the previous release](https://github.com/nvaccess/nvda/compare/release-2024.2...docsFor2024.3?expand=1)
- Comparison command:
`git diff release-2024.2...docsFor2024.3  -- "**/en/*.md" "**/developerGuide.md"`

Common mistakes to check for:
- Issue/PR reference in changes file is incorrect
- Incorrect spelling.
- Shortcuts added without code markdown (e.g. `NVDA+d`)
- List items not indented by a multiple of 2 spaces (regex `^ (  )*-`)
- Double spaces (regex `[^ ]  `)
- Inconsistent use of single vs double quote. (regex `' `)

## Summary by CodeRabbit

- **New Features**
  - Added support for Help Tech Activator Pro displays.
  - Introduced new key commands and notification for add-on updates on NVDA startup.

- **Enhancements**
  - Updated dependencies: LibLouis, fast_diff_match_patch, NSIS, markdown, and nh3.
  - Enhanced Unicode normalization and Unicode CLDR.

- **Bug Fixes**
  - Various bug fixes improving overall stability.

- **Documentation**
  - Updated user guide with new configuration options for NVDA modifier keys.
  - Added detailed information about tables in NVDA add-ons to the developer guide.

### CODE REVIEW

**Review by Qchristensen — COMMENTED** [review]
All looks good.

**Review by Qchristensen — APPROVED** [review]

**Review by CyrilleB79 — COMMENTED** [review]
Here is my review.

In addition to the comments in the code, the following points should be looked at in the change log:
1. "menuitems": rather use "menu items" with dash
2. Remove the following sentence which is useless since there are no deprecations:
   `Please refer to [the developer guide](https://www.nvaccess.org/files/nvda/documentation/developerGuide.html#API) for information on NVDA's API deprecation and removal process.`
   Note that this has not been removed for 2024.2, but IMO, it was a mistake.

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -23,12 +27,11 @@ Unicode CLDR has also been updated.
   * This can be useful when reading characters that are unknown to a particular speech synthesizer or braille table and which have a compatible alternative, like the bold and italic characters commonly used on social media.
   * It also allows reading of equations in the Microsoft Word equation editor. (#4631)
   * You can enable this functionality for both speech and braille in their respective settings categories in the NVDA Settings dialog.
-* New Braille features:
-  * Added support for the Help Tech Activator Pro displays. (#16668)
 * By default, after NVDA startup, you will be notified if any add-on updates are available. (#15035)
   * This can be disabled in the "Add-on Store" category of settings.
   * NVDA checks daily for add-on updates.
   * Only updates within the same channel will be checked (e.g. installed beta add-ons will only notify for updates in the beta channel).
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply user_docs/en/changes.md:33]: I find sublists easier to read than large paragraphs for items. The change log is in dot point style, which in my experience prefers brevity and subpoints. This tends to be more the common style now in our change log.

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -39,20 +42,22 @@ Unicode CLDR has also been updated.
...
+* By default, a new folder will be created when making a portable copy.
+Warnings have been added when writing to a non-empty directory. (#16684)
 
 ### Bug Fixes
+
 * Windows 11 fixes:
   * NVDA will no longer appear to get stuck when closing the clipboard history and emoji panel. (#16346, #16347, @josephsl)
-  * NVDA will announce visible candidates again when opening the Windows 11 IME interface. (#14023, @josephsl)
+  * NVDA will announce visible candidates again when opening the IME interface. (#14023, @josephsl)
   * NVDA will no longer announce "clipboard history" twice when navigating through the emoji panel menu items. (#16532, @josephsl)
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply user_docs/en/changes.md:65]: Seems accurate to the original issue: #16532

[the remark this answers is not in the document]
Comment (CyrilleB79) [reply user_docs/en/changes.md:65]: The title of the original issue mentions emoji panel, but the STR only mentions Clipboard history panel and its shortcut (`windows+V`).
That's why I was assuming that there were a cfusion somewhere and that I had tagged @josephsl for checking.

**Pull request merged**